tasks/verifySpokePool.ts

```diff
@@ -0,0 +1,93 @@
+      assert(bytestring.slice(2, 26) === "000000000000000000000000");
+      return ethers.utils.getAddress(bytestring.slice(0, 2).concat(bytestring.slice(-40)));
+    };
+
+    const spokeChainId = parseInt(await getChainId());
+    const hubChainId = Object.values(MAINNET_CHAIN_IDs).includes(spokeChainId) ? MAINNET : SEPOLIA;
+
+    const spokeAddress = getDeployedAddress("SpokePool", spokeChainId, true);
+    const hubAddress = getDeployedAddress("HubPool", hubChainId, true);
+
+    const provider = new ethers.providers.StaticJsonRpcProvider(network.config.url);
+
+    // Initialize contracts. Only generic SpokePool functions are used, so Ethereum_SpokePool is OK.
+    const { abi } = await deployments.get("Ethereum_SpokePool");
+    const spokePool = new ethers.Contract(spokeAddress, abi, provider);
+
+    const spokeFunctions = [
+      "chainId",
+      "getCurrentTime",
+      "depositQuoteTimeBuffer",
+      "fillDeadlineBuffer",
+      "wrappedNativeToken",
+      // "withdrawalRecipient",
+      "crossDomainAdmin",
+    ] as const;
+    const multicall = await spokePool.callStatic.multicall(
+      spokeFunctions.map((fn) => spokePool.interface.encodeFunctionData(fn))
+    );
+    const results = Object.fromEntries(spokeFunctions.map((fn, idx) => [fn, multicall[idx]]));
+
+    // Log state from SpokePool
+    const originChainId = Number(results.chainId);
+    console.log("SpokePool.chainId()".padEnd(TEXT_PADDING) + ": " + originChainId);
+    assert(Number(originChainId) === spokeChainId, `${originChainId} != ${spokeChainId}`);
+
+    const currentTime = Number(results.getCurrentTime);
+    const formattedTime = `${currentTime} (${new Date(Number(currentTime) * 1000).toUTCString()})`;
+    console.log("SpokePool.getCurrentTime()".padEnd(TEXT_PADDING) + ": " + formattedTime);
+
+    const quoteTimeBuffer = Number(results.depositQuoteTimeBuffer);
+    console.log("SpokePool.depositQuoteTimeBuffer()".padEnd(TEXT_PADDING) + ": " + quoteTimeBuffer);
+
+    const fillDeadlineBuffer = Number(results.fillDeadlineBuffer);
+    console.log("SpokePool.fillDeadlineBuffer()".padEnd(TEXT_PADDING) + ": " + fillDeadlineBuffer);
+
+    const wrappedNative = bytes32ToAddress(results.wrappedNativeToken);
+    const wrappedNativeSymbol = spokeChainId === POLYGON ? "WMATIC" : "WETH";
+    const expectedWrappedNative = TOKEN_SYMBOLS_MAP[wrappedNativeSymbol].addresses[spokeChainId];
+    assert(wrappedNative === expectedWrappedNative, `wrappedNativeToken: ${wrappedNative} != ${expectedWrappedNative}`);
+    console.log("SpokePool.wrappedNativeToken()".padEnd(TEXT_PADDING) + ": " + wrappedNative);
+
+    //    const withdrawalRecipient = bytes32ToAddress(results.withdrawalRecipient);
+    //    console.log("SpokePool.withdrawalRecipient()".padEnd(TEXT_PADDING) + ": " + withdrawalRecipient);
+    //    assert(withdrawalRecipient === hubAddress, `withdrawalRecipient: ${withdrawalRecipient} != ${hubAddress}`);
```

Comment:
This is currently commented out because all existing SpokePool deployments have `hubPool` but in this branch it has been renamed to `withdrawalRecipient`.